### ISSUE TITLE
[Android/iOS] setFormFieldValue and saveCurrentDocument should return promise

### ISSUE THREAD

**Issue by irgendeinich** [PSPDFKit/react-native#296]
# Details

RIght now when calling `setFormFieldValue` or `saveCurrentDocument` there is no way to know if the operation is already done, this leads to issues if you want to set a bunch of form fields and then afterward save the document.

This happens because (at least on Android) `setFormFieldValue` works asynchronously and as such if you have code like:

```
this.refs.pdfView.setFormFieldValue("Field1", "Value1");
this.refs.pdfView.setFormFieldValue("Field2", "Value2");
this.refs.pdfView.setFormFieldValue("Field3", "Value3");
this.refs.pdfView.saveCurrentDocument(); 
```

The saving will happen before the values are set and the values won't be saved. You also can't do an action after a manually triggered save completed unless you also set `onDocumentSaved` on the `PSPDFKitView`.

# Acceptance Criteria

- [x] `setFormFieldValue` and `saveCurrentDocument` return promise on Android.
- [x] `setFormFieldValue` and `saveCurrentDocument` return promise on iOS.

**Comment by nickwinder** [comment]
UWP already returns a promise for `saveCurrentDocument `. https://github.com/PSPDFKit/react-native/blob/c1a330e3152945487d0eda663ba6936aa05adfd8/index.windows.js#L112
But `setFormFieldValue` is not implemented at all on UWP...

**Comment by irgendeinich** [comment]
Resolved with #308
